src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -767,6 +774,19 @@ define('diagramStore', ['jquery', 'xwiki-meta', 'xwiki-utils', 'draw.io', 'xwiki
     oldImportFiles.apply(this, newArgs);
   }
 
+  var originalLoadUrl = EditorUi.prototype.loadUrl;
+  EditorUi.prototype.loadUrl = function(url, success, error, forceBinary, retry, dataUriPrefix) {
+    let exportedUrl = diagramUtils.getParameterValueFromURL('url', url);
+    if (exportedUrl) {
+      let diagramXML = diagramUtils.getDiagramXMLFromURL(exportedUrl);
+      if (diagramXML) {
+        success(diagramXML);
+      }
```

Comment:
If ``diagramXML`` is null (can't be extracted from the URL) then the callback (success) is not called, which means the diagram editor is left waiting. We need to either call ``success`` or ``error`` ourselves or fallback on the original behavior (``originalLoadUrl``). The later option (``originalLoadUrl``) is probably the safest.